### DISCUSSION TITLE
Getting Unable to compile federated types, Error: compile TS failed error when using module federation

### DISCUSSION THREAD

**Discussion by AsuraKev** [web-infra-dev/rsbuild#3569] [Q&A]
Hi there, I am trying out the module federation plugins with Rsbuild but the expose method is triggering the error

**Unable to compile federated types, Error: compile TS failed, the original command is 'npx tsc --project D:\Personal Projects\MicroFrontend\myMask Provider MFE\node_modules\.federation\tsconfig.93662bdac7c7c203575a7e5eb91a23a4.json'
Error: ENOENT: no such file or directory, open 'D:\Personal Projects\MicroFrontend\myMask Provider MFE\dist\@mf-types.zip'**

rsbuild.config.ts

![image](https://github.com/user-attachments/assets/6778853f-5e09-4df1-987e-ae462fefe6e5)

bootstrap.tsx

![image](https://github.com/user-attachments/assets/858229e9-8815-4cce-aefd-648a91a36164)

any idea whats the causes? so far its pretty much just standard setup. thanks!

**Comment by shallow-alchemy** [comment]
I had the same issue. Using the output that tells you the original `npx tsc` command, you can debug what is failing in the typescript build. Resolving the build errors should enable the correct types to be generated
